### ISSUE TITLE
[🐛BUG] run_hyper 使用多组参数进行hyper_tuning时，NGCF模型卡住0%长时间不动。

### ISSUE THREAD

**Issue by GitHub-ZHY** [RUCAIBox/RecBole#1221]
**描述这个 bug**
run_hyper 使用多组参数进行hyper_tuning时，NGCF模型卡住0%长时间不动

**如何复现**
复现这个 bug 的步骤：
1. 您引入的额外 yaml 文件
ngcf.yaml:
embedding_size: 64
hidden_size_list: [64,64,64]
node_dropout: 0.0
message_dropout: 0.1
epoch: 20
--------------------------------------------------------
hyper.test:
learning_rate choice [0.01,0.005,0.001,0.0005,0.0001]
hidden_size_list choice ['[64,64,64]','[128,128,128]','[256,256,256]','[512,512,512]']
node_dropout choice [0.0,0.1,0.2]
message_dropout choice [0.0,0.1,0.2,0.3]
reg_weight choice [1e-5,1e-4]
delay choice [1e-5,1e-4,1e-4,1e-2,1e-1]
--------------------------------------------------------
2. 您的代码

3. 您的运行脚本
python run_hyper.py --model=NGCF --dataset=ml-1m --config_files=NGCF.yaml --params_file=hyper.ngcf

**预期**
能跑起来

**屏幕截图**
running parameters:
{'delay': 1e-05, 'hidden_size_list': '[64,64,64]', 'learning_rate': 0.005, 'message_dropout': 0.3, 'node_dropout': 0.0, 'reg_weight': 1e-05}
  0%|          | 0/2400 [00:00<?, ?trial/s, best loss=?]

**链接**
nil

**实验环境（请补全下列信息）：**
 - 操作系统: [Windows]
- RecBole 版本 [0.1.0]
 - Python 版本 [3.8]
- PyTorch 版本 [如 1.8.1]
- cudatoolkit 版本 [11.4]

**Comment by Sherry-XLL** [comment]
@GitHub-ZHY 你好，进度条卡住不动可能与你的参数配置和实验设备有关。

在设备一定的情况下，模型的运行时间很大程度上取决于模型的复杂度、设置的批处理大小和收敛轮数。NGCF 模型是非线性的图神经网络模型，训练相对较慢；同时在默认情况下使用的是全排序进行评测，评测时占用的时间较多。由于你的配置文件中没有设置训练和验证的相关参数，所以调参时仍使用的是默认的 `train_batch_size = 2048` 和 `eval_batch_size = 4096`，效率较低。

为了提升效率，模型运行时可以在配置文件中设置较大的 `train_batch_size` 和 `eval_batch_size` 如下：

```
train_batch_size: 4096
eval_batch_size: 40960000
```

另外，你的配置文件中设置 `epoch: 20` 大概是为了测试轮数较小情况下的运行情况，但是控制总轮数的参数为 `epochs`，参数名称设置不当使得默认的 `epochs` 仍为 300。

为了预估超参数微调的时间，建议先使用 `python run_recbole.py --model=NGCF --dataset=ml-1m --config_files=NGCF.yaml` 命令查看模型在一组参数下的运行时间，再有针对性地选择参数范围进行调节。

**Comment by GitHub-ZHY** [comment]
好的，谢谢您的指点，多谢！

**Comment by Sherry-XLL** [comment]
@GitHub-ZHY 如果运行 `run_recbole.py` 的话 `get_gpu_usage` 会显示 GPU 的显存使用量，但 RecBole 并没有提供参数来调整 GPU 显存的使用量。
